src/improvements/template/L1PortalExecuteL2Call.sol

```diff
@@ -0,0 +1,117 @@
+    uint64 public gasLimit; // L2 gas limit (required)
+    bool public isCreation; // Whether to create a contract on L2 (defaults to false)
+
+    /// @notice Default Safe name. Can be overridden via `safeAddressString` in config.toml.
+    function safeAddressString() public pure override returns (string memory) {
+        return "ProxyAdminOwner";
+    }
+
+    /// @notice The contracts expected to have storage writes during execution.
+    /// Allowlist the OptimismPortal since it will mutate state (queue/event) on deposit.
+    function _taskStorageWrites() internal pure override returns (string[] memory) {
+        string[] memory _storageWrites = new string[](1);
+        _storageWrites[0] = "OptimismPortal";
+        return _storageWrites;
+    }
+
+    /// @notice The contracts expected to have balance changes during execution.
+    /// Allowlist the OptimismPortal to receive ETH (value) in the deposit call.
+    function _taskBalanceChanges() internal pure override returns (string[] memory) {
+        string[] memory _balanceChanges = new string[](1);
+        _balanceChanges[0] = "OptimismPortal";
+        return _balanceChanges;
+    }
+
+    /// @notice Parse config and initialize template variables.
+    /// Expected TOML keys:
+    /// - portal: address (L1 OptimismPortal) OR addresses.OptimismPortal in [addresses]
+    /// - l2Target: address (L2 target address)
+    /// - l2Data: hex string (e.g. 0x1234...)
+    /// - gasLimit: uint (will be cast to uint64)
+    /// - value: uint (optional, default 0)
+    /// - isCreation: bool (optional, default false)
+    function _templateSetup(string memory _taskConfigFilePath, address) internal override {
+        string memory _toml = vm.readFile(_taskConfigFilePath);
+
+        // Resolve portal from registry first if available, else read explicit field.
+        try simpleAddrRegistry.get("OptimismPortal") returns (address p) {
+            portal = payable(p);
+        } catch {
+            portal = payable(_toml.readAddress(".portal"));
+        }
+        require(portal != address(0), "portal must be set (addresses.OptimismPortal or .portal)");
+
+        l2Target = _toml.readAddress(".l2Target");
+        require(l2Target != address(0), "l2Target must be set");
+
+        // Read hex string and parse to bytes.
+        string memory _dataHex = _toml.readString(".l2Data");
+        l2Data = vm.parseBytes(_dataHex);
+        require(l2Data.length > 0, "l2Data must be set");
+
+        uint256 _gasLimitTmp = _toml.readUint(".gasLimit");
+        require(_gasLimitTmp > 0 && _gasLimitTmp <= type(uint64).max, "invalid gasLimit");
+        gasLimit = uint64(_gasLimitTmp);
+
+        // Optional fields
+        valueWei = 0;
+        try vm.parseTomlUint(_toml, ".value") returns (uint256 _v) {
+            valueWei = _v;
+        } catch {}
+
+        isCreation = false;
+        try vm.parseTomlBool(_toml, ".isCreation") returns (bool _b) {
+            isCreation = _b;
```

Comment:
The portal has this check so it's arguably unnecessary, but might be better UX to check and revert here if `isCreation && l2Target != address(0)`. Up to you